pandas/core/nanops.py

```diff
@@ -637,7 +619,7 @@ def _maybe_null_out(result, axis, mask):
             else:
                 result = result.astype('f8')
             result[null_mask] = np.nan
-    else:
+    elif result is not tslib.NaT:
```

Comment:
no I mean result should already be an int if it's M8 as wrapping is the last step